File: Client-Leger/app/src/main/java/com/zouatene/colorimage/model/drawelement/Line.kt
package com.zouatene.colorimage.model.drawelement

data class Line(
    override var id:String,
    override var z: Int,
    override var matrix: IMatrix?,
    override var strokeWidth: Float,
    override var isSelected: Boolean,
    var color: String?,
    val points: MutableList<Point>?,
     ) : VectorObject()

File: Serveur/classes/shapes/line.ts
import { Line } from "../../interfaces/drawing/line";
import { Point } from "../../interfaces/drawing/point";
import { VectorObjectClass } from "./vectorObject";

export class LineClass extends VectorObjectClass {

    private points: Array<Point>;

    constructor(
        newLine: Line
    ) {
        super(
            newLine.id,
            newLine.isSelected,
            newLine.z,
            newLine.strokeWidth,
            newLine.color,
            newLine.matrix);
        this.points = newLine.points;
    }

    public getPoints(): Array<Point> {
        return this.points;
    }

    public addPoints(point: Point): void {
        this.points.push(point);
    }
}